src/parse.rs

```diff
@@ -500,10 +518,14 @@ impl<'a> Bytes<'a> {
     }
 
     pub fn eat_byte(&mut self) -> Result<u8> {
-        let peek = self.peek_or_eof()?;
-        let _ = self.advance_single();
+        let peek = self.peek_byte_or_eof()?;
+        if peek.is_ascii() {
+            let _ = self.advance_byte();
 
-        Ok(peek)
+            Ok(peek)
+        } else {
+            Err(Error::NoCharBoundary)
+        }
     }
 
     pub fn expect_byte(&mut self, byte: u8, error: Error) -> Result<()> {
```

Comment:
Do we still actually need this method? Would it be possible to eat (and expect) `char`s instead?

Reply:
We probably don't, it might pose a marginal performance benefit, but probably doesn't matter.